file/test/sentry_file_test.dart

```diff
@@ -657,6 +657,286 @@ void main() {
       );
     });
   });
+
+  group('$SentryFile span sets `blocked_main_thread`', () {
```

Comment:
All tests are using the `isMainIsolate` fake but not starting a real background isolate and reading a file from it to verify that indeed returns `true` or `false` correctly.
So it is hard to tell that the solution works.
I'd write a test that creates a background isolate and wait for its result, and asserting that the span contains the right result.
This test could be done on the flutter integration test for example if needs to be, or even the example app, since there the flutter min. version is higher or can be higher

Reply:
Added two tests to the example app. The first runs on the main isolate and the second init's sentry only on a bg isolate and runs there. Results as expected, thx for the suggestion! 👍